File: testsuites/LLT/compiler/CJMP/java/integration/mirror-prop/src/java/com/java/lib/M.java
// Copyright (c) Huawei Technologies Co., Ltd. 2025. All rights reserved.
// This source file is part of the Cangjie project, licensed under Apache-2.0
// with Runtime Library Exception.
//
// See https://cangjie-lang.cn/pages/LICENSE for license information.

package com.java.lib;

public class M {
    public final long x;
    public long xm;

    public static final long xs = 5;
    public static long xms = xs;
    
    public final M m;
    public M mm;
    public static final M ms = new M(5);
    public static M mms = ms;

    public M(long i) {
        System.out.println("M(" + i + ")");
        this.x = i;
        this.xm = i;

        this.m = this;
        this.mm = this;
    }
}

File: testsuites/LLT/compiler/CJMP/objc/integration/objc_impl/field/src/objc/M.m
// Copyright (c) Huawei Technologies Co., Ltd. 2025. All rights reserved.
// This source file is part of the Cangjie project, licensed under Apache-2.0
// with Runtime Library Exception.
//
// See https://cangjie-lang.cn/pages/LICENSE for license information.

#import "M.h"

@implementation M

- (id)init {
    if (self = [super init]) {
        printf("objc: [M init]\n");
    }

    return self;
}

- (void) identity {
    printf("objc: I'm a class M instance\n");
}

@end

@implementation M1

- (id)init {
    if (self = [super init]) {
        printf("objc: [M1 init]\n");
    }

    return self;
}

- (void) identity {
    printf("objc: I'm a class M1 instance\n");
}

@end
